src/sched.rs

```diff
@@ -215,6 +215,112 @@ mod sched_linux_like {
 
         Errno::result(res).map(drop)
     }
+
+    libc_bitflags! {
+        pub struct SchedFlags: c_int {
```

Comment:
This isn't a straightforward bitfield, and treating it as such is causing incorrect matches. Instead, the entries aside from `SCHED_RESET_ON_FORK` are a standard enum, but the enum can be ORed with `SCHED_RESET_ON_FORK`. Need to figure out how to represent this.

Reply:
@asomers I am revisiting this and would like some input on how to best expose this type in Rust. The lower bits of the type in libc are effectively an enum, and the upper bits are a bitfield.

I've considered the following options:
  1. Create 4-byte `SchedType` enum and `SchedFlags` bitfield types, and have the user pass them separately. nix can then OR them together internally, and hide this as an implementation detail.
  2. Same as 1., but create an 8-byte struct to hold these two fields.
  3. Create 2-byte `SchedType` enum and `SchedFlags` bitfield types, and have the user pass them separately. nix can then combine them via bitshifting to form a 4-byte value, and hide this as an implementation detail.
  4. Same as 3., but create a 4-byte struct to hold these two fields.
  5. Create a completely custom 4-byte type with accessors to properly manipulate the data.


--- 

1\. and 2. are wasteful in terms of space, but don't make assumptions about the number of bits assigned to the enum and flag portions of the type.

3\. and 4. place restrictions on the number of future bits in the enum and flag portions of the type, but aren't wasteful in terms of space.

5\. is the most work, and adds complexity to using the scheduler functions.


Thoughts?

Reply:
1 and 3 sound the most ergonomic.  I wouldn't worry about the size of the field, because Nix doesn't currently guarantee ABI stability across releases, and the field's size won't be expressed in the API at all.  But there is another important question: do any of the bitfield-like bits apply on non-Linux operating systems?  If not, then options 1-4 would seem awfully odd on non-Linux platforms.  5 would be the best in that case.

Reply:
The bitfield-like bits seem to only apply on Linux and Android (so... Linux). 

Does it make sense to wrap them in a struct (so 2 or 4), but have the second member of the struct conditionally defined only on Linux and Android? I.e. the struct would have a single enum member on non-Linux platforms, and an enum and bitfield on Linux + Android. Maybe that's just 5.

Reply:
Yeah, I like that idea.  Even better if the struct's most common contructors don't involve the bitffield-like bits, making them portable.

Reply:
I was working on implementing these changes, but realized I wouldn't be satisfied with the API if it didn't support at least one platform without `SCHED_RESET_ON_FORK`. I picked FreeBSD (and will probably try to pick up other Tier 1 targets provided I can validate them locally), but have now run into the issue of libc not being up-to-date for some of these functions + platforms.

I'm working on some contributions to libc to expand support for `sched_*` functions, and will get back to this PR as those changes are submitted.

Reply:
@asomers I've put up the changes implementing this new approach; I don't love the type names, but I couldn't come up with anything better. Ditto for the namespace.

I kept this new commit separate for now, but will squash before this is actually submitted.

The dependency on libc was bumped to pick up the following changes:
 - https://github.com/rust-lang/libc/commit/b68bd77e3130115a5312d856ce881e0ec8533b0e
 - https://github.com/rust-lang/libc/commit/f148bfd673823b07bb7bdee7369cb0f88d368f5d
 - https://github.com/rust-lang/libc/commit/10eff9941f3002f31e18a2befe4bfcb86868140e
 - https://github.com/rust-lang/libc/commit/3c17b02bbec9a2db73cff14b225f3627bf07ac0a

Reply:
Bumped libc again to pick up https://github.com/rust-lang/libc/commit/0904320c928ec98b3afea7874b29c7902490c836.